test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs

```diff
@@ -5711,5 +5711,486 @@ public void Dotnet_New_Template_Restore_Pack_Success(string template)
+                result.AllOutput.Should().NotContain(NuGetLogCode.NU5104.ToString());
+
+                using (var nupkgReader = new PackageArchiveReader(nupkgPath))
+                {
+                    var nuspecReader = nupkgReader.NuspecReader;
+
+                    var dependencyGroups = nuspecReader
+                        .GetDependencyGroups()
+                        .OrderBy(x => x.TargetFramework,
+                            new NuGetFrameworkSorter())
+                        .ToList();
+
+                    Assert.Equal(1,
+                        dependencyGroups.Count);
+
+                    var dependencyPackage = dependencyGroups[0].Packages.ToList();
+                    Assert.Equal(1, dependencyPackage.Count);
+                    Assert.Equal(prereleaseDependencyName, dependencyPackage[0].Id);
+                    Assert.Equal(new VersionRange(new NuGetVersion(prereleaseDependencyVersion), true, null, true), dependencyPackage[0].VersionRange);
+                    Assert.Equal(new List<string> { "Analyzers", "Build" }, dependencyPackage[0].Exclude);
+                    Assert.Empty(dependencyPackage[0].Include);
+                }
+            }
+        }
+
+        [Fact]
+        public async Task PackCommand_PackProject_PackageReference_PreReleaseDependencies_PartialWarningSuppressed_Fails()
+        {
+            using (var pathContext = msbuildFixture.CreateSimpleTestPathContext())
+            {
+                var prereleaseDependencyAName = "PreReleasePackageA";
+                var prereleaseDependencyAVersion = "4.8.0-beta00011";
+                var prereleaseDependencyAPackage = new SimpleTestPackageContext(prereleaseDependencyAName, prereleaseDependencyAVersion);
+                prereleaseDependencyAPackage.Files.Clear();
+                prereleaseDependencyAPackage.AddFile("_._");
+                await SimpleTestPackageUtility.CreatePackagesAsync(pathContext.PackageSource, prereleaseDependencyAPackage);
+
+                var prereleaseDependencyBName = "PreReleasePackageB";
+                var prereleaseDependencyBVersion = "4.4.0-preview1-25305-02";
+                var prereleaseDependencyBPackage = new SimpleTestPackageContext(prereleaseDependencyBName, prereleaseDependencyBVersion);
+                prereleaseDependencyBPackage.Files.Clear();
+                prereleaseDependencyBPackage.AddFile("_._");
+                await SimpleTestPackageUtility.CreatePackagesAsync(pathContext.PackageSource, prereleaseDependencyBPackage);
+
+                var projectName = "ClassLibrary1";
+                string testDirectory = pathContext.WorkingDirectory;
+                var workingDirectory = Path.Combine(testDirectory, projectName);
+                var projectFile = Path.Combine(workingDirectory, $"{projectName}.csproj");
+                msbuildFixture.CreateDotnetNewProject(testDirectory, projectName);
+                string projectXml = $@"<Project Sdk=""Microsoft.NET.Sdk"">
```

Comment:
There are other utilities that simply edit the PackageReference.

It'd make your test more future proof. 

You're also adding TreatWarningsASErrors in the projects, which is not something obvious from the name.

Reply:
I received [another feedback](https://github.com/NuGet/NuGet.Client/pull/4369#discussion_r769896727) contradict this.
I removed TreatWarningsASErrors from this tests, but I renamed another test to `PackCommand_MultiTfm_PrereleaseDependency_TreatWarningsAsErrors_FailsAndLogsWarning` so it would be more obvious why it's failing.

Reply:
I very much disagree with feedback like that @zivkan. 

You don't have to change it, but we shouldn't *hardcode* things, to allow for the template to mutate so that we're actually testing with the right thing.

Reply:
Resolving this thread.

Reply:
@nkolev92 the reason I suggested that is because tests should be EASY to read. The easier to read the easier is it to validate that the test is validating the right thing, and therefore that the production code is doing the right thing. Therefore, having stupidly simple tests doing `var x = 2; var result = Class.DoThing(x); Assert.Equals(4, result);` is the dream. On the other hand, having tens of lines of code to read the XML of a project file, do XML navigation to find a specific element, then add a new PackageReference element, then save the file is much harder to read and understand than a hardcoded SDK-style project file. If we were testing non-SDK style projects, then I would agree, because the code to modify those project files would be less than a full project file.

This repo is full of tests that do a lot more than is necessary for the thing that's being tested, because tests are so long and difficult to write, so many people copy-paste existing tests, change just a little bit and commit, without removing the parts unrelated to what the test is trying to validate. That means it's hard for us reviewing the PR because there's more to read, and then it's harder to validate whether it's doing the right thing.

tl;dr helper methods that don't make test code shorter and easier to understand are an anti-pattern.  Writing tests with the ideal of being flexible, but harder to read is not good.

Reply:
Inconsistency is hard to read. 

> This repo is full of tests that do a lot more than is necessary for the thing that's being tested, because tests are so long and difficult to write, so many people copy-paste existing tests

Editing the project file avoids overspecifying.

I don't know how you arrived to an anti-pattern, but I certainly don't want to start a big discussion about test code changes.